synapse/cores/storage.py

```diff
@@ -0,0 +1,827 @@
+        self._core_xacts = {}
+
+        # Dicts for storing retrieval methods
+        self.sizebymeths = {}
+        self.rowsbymeths = {}
+
+        # Register handlers for lifting rows/siexes
+        self.initRowsBy('gt', self.rowsByGt)
+        self.initRowsBy('lt', self.rowsByLt)
+        self.initRowsBy('ge', self.rowsByGe)
+        self.initRowsBy('le', self.rowsByLe)
+        self.initRowsBy('range', self.rowsByRange)
+
+        self.initSizeBy('ge', self.sizeByGe)
+        self.initSizeBy('le', self.sizeByLe)
+        self.initSizeBy('range', self.sizeByRange)
+
+        # Events for handling savefile loads/saves
+        self.loadbus.on('core:save:add:rows', self._loadAddRows)
+        self.loadbus.on('core:save:del:rows:by:iden', self._loadDelRowsById)
+        self.loadbus.on('core:save:del:rows:by:prop', self._loadDelRowsByProp)
+        self.loadbus.on('core:save:set:rows:by:idprop', self._loadSetRowsByIdProp)
+        self.loadbus.on('core:save:del:rows:by:idprop', self._loadDelRowsByIdProp)
+        self.loadbus.on('syn:core:blob:set', self._onSetBlobValu)
+        self.loadbus.on('syn:core:blob:del', self._onDelBlobValu)
+
+        # Cache blob save mesgs which may be fired during storage layer init
+        _blobMesgCache = []
+        self.savebus.on('syn:core:blob:set', _blobMesgCache.append)
+        self.savebus.on('syn:core:blob:del', _blobMesgCache.append)
+
+        # Perform storage layer initializations
+        self._initCoreStor()
+
+        # Disable the blob message caching
+        self.savebus.off('syn:core:blob:set', _blobMesgCache.append)
+        self.savebus.off('syn:core:blob:del', _blobMesgCache.append)
+
+        # process a savefile/savefd if we have one (but only one of the two)
+        savefd = self._link[1].get('savefd')
+        if savefd is not None:
+            self.setSaveFd(savefd)
+        else:
+            savefile = self._link[1].get('savefile')
+            if savefile is not None:
+                savefd = s_common.genfile(savefile)
+                self.setSaveFd(savefd, fini=True)
+
+        # The storage layer initialization blob events then trump anything
+        # which may have been set during the savefile load and make sure they
+        # get saved as well
+        if 'savefd' in link[1] or 'savefile' in link[1]:
+            for evtname, info in _blobMesgCache:
+                self.savebus.fire(evtname, **info)
+                self.loadbus.fire(evtname, **info)
+
+        if not self.hasBlobValu('syn:core:created'):
+            self.setBlobValu('syn:core:created', s_common.now())
+
+        self.onfini(self._finiCoreStore)
+
+    @staticmethod
+    @s_config.confdef(name='storage')
+    def _storage_confdefs():
+        confdefs = (
+            ('rev:storage', {'type': 'bool', 'defval': 1, 'doc': 'Set to 0 to disallow storage version updates'}),
+        )
+        return confdefs
+
+    # Handlers which should be untouched!!!
+    def register_cortex(self, core):
```

Comment:
It feels like getTufosBy() would act as a bit of a passthrough to getJoinBy, which would in turn end up calling rowsByMeths functions.  Will have to experiment with this though - this can now be explored since we have unit tests for all of the getTufosBy handlers.